src/CoinLockerSystem.java

```
@@ -134,24 +103,125 @@ private Locker getAvailableLocker(LockerSize size) {
                 if (smallLockers.size() > 0) {
                     locker = smallLockers.get(0);
                     smallLockers.remove(locker);
+                    break;
                 }
-                break;
             case MEDIUM:
                 if (mediumLockers.size() > 0) {
                     locker = mediumLockers.get(0);
                     mediumLockers.remove(locker);
+                    break;
                 }
-                break;
             case LARGE:
                 if (largeLockers.size() > 0) {
                     locker = largeLockers.get(0);
                     largeLockers.remove(locker);
+                    break;
                 }
-                break;
         }
-        if(locker != null){
+        if (locker != null) {
```

What if we were not able to find any matching locker? I think we should throw an exception here and add matching unit test.

---

Revised